### DISCUSSION TITLE
Client crashing when connection is cut while spectating in a mp lobby

### DISCUSSION THREAD

**Discussion by nzxl101** [ppy/osu#16821] [Q&A] [answered]
Should be expected, but could be handled better imo.
Back button gives an error and after clicking for 2-3 times the client just decides to crash.

Steps to reproduce:
1. Join a multiplayer lobby
2. Spectate
3. Cut internet connection

https://user-images.githubusercontent.com/63413558/152900105-f78ccccb-f692-4d38-af69-b6c924f3cd6a.mp4

[runtime.log](https://github.com/ppy/osu/files/8019894/runtime.log)
[network.log](https://github.com/ppy/osu/files/8019895/network.log)

**Comment by smoogipoo** [comment, marked as answer]
https://github.com/ppy/osu/issues/16822
